### PR TITLE
fix: JDT spoon compiler builds imports and then comments

### PR DESCRIPTION
**Pull request by pvojtechovsky**
It allows to assign comments on the imports

```java
//comment
import a.b.C;
```
The test will be in #2683.

It is ready for merge

### CODE REVIEW

File: src/main/java/spoon/support/compiler/jdt/JDTBasedSpoonCompiler.java [hunk 1]
```diff
@@ -424,53 +425,55 @@ protected boolean buildUnitsAndModel(JDTBuilder jdtBuilder, SpoonFolder sourcesF
...
-			getEnvironment().getSpoonProgress().start(SpoonProgress.Process.MODEL);
-		}
 		JDTTreeBuilder builder = new JDTTreeBuilder(factory);
 		List<CompilationUnitDeclaration> unitList = this.sortCompilationUnits(units);
 
+		forEachCompilationUnit(unitList, SpoonProgress.Process.MODEL, unit -> {
+			unit.traverse(builder, unit.scope);
+		});
+		if (getFactory().getEnvironment().isAutoImports()) {
+			// we need first to go through the whole model before getting the right reference for imports
```
Comment (surli) [inline src/main/java/spoon/support/compiler/jdt/JDTBasedSpoonCompiler.java]: I think you misplaced the comments in this method

**Comment by surli** [comment]
Nice refactoring!

**Comment by pvojtechovsky** [comment]
comments are fixed now

**Comment by monperrus** [comment]
Thanks a lot!

**Pull request merged**